### PR TITLE
Azure-support fixes

### PR DESCRIPTION
**Pull request by SauarvD**
### Requirements

* Fixes for azure-support branch after syncing with develop

### Description of the Change

Jazz developers to use azure as an extension for their serverless apps!

### CODE REVIEW

File: core/jazz_ui/src/app/pages/services-list/services-list.component.ts [hunk 1]
```diff
@@ -176,6 +175,9 @@ private intervalSubscription: Subscription;
         id: service.id,
         data: service
       };
+      if(service.deployment_accounts){
+        serviceRow['platform'] = service.deployment_accounts[0].provider
+      }
```
Comment (raghits) [inline core/jazz_ui/src/app/pages/services-list/services-list.component.ts]: Do we need a default here?

File: builds/jenkins-build-pack-api/Jenkinsfile [hunk 2]
```diff
@@ -425,10 +425,6 @@ node() {
                   error "Deployment targets are not defined"
               }
 
-              if (domain != "jazz") {
```
Comment (devsatishm) [inline builds/jenkins-build-pack-api/Jenkinsfile]: Why is this removed?

File: builds/service-onboarding-build-pack/Jenkinsfile [hunk 3]
```diff
@@ -141,11 +142,17 @@ node  {
 
 	stage('Get Service Template'){
 		try {
+			try {
+				sh 'rm -rf *'
```
Comment (devsatishm) [inline builds/service-onboarding-build-pack/Jenkinsfile]: `rm -rf *` should just do it. You don't need the next cmd.

File: core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts [hunk 4]
```diff
@@ -122,9 +121,15 @@ export class ServiceDetailComponent implements OnInit {
...
-        created_by: service.created_by,
-        accountID: service.deployment_accounts[0].accountId,
-        region: service.deployment_accounts[0].region
+        created_by: service.created_by
+      }
+      if(service.deployment_accounts){
+        returnObject['accountID'] = service.deployment_accounts[0].accountId,
+        returnObject['region'] = service.deployment_accounts[0].region
+      } else {
+        // If no deployment_accounts is present, then fetch from default aws region and account, otherwise fetch from provider (aws/azure) specific values. We are not assigning any default azure values.
```
Comment (devsatishm) [inline core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts]: Nope, this is not needed. Every service should have deployment_accounts with accountId & region. Remove the `else` block.

File: core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts [hunk 5]
```diff
@@ -219,7 +224,12 @@ export class ServiceDetailComponent implements OnInit {
     this.isLoadingService = true;
     this.http.get('/jazz/services/' + id, null, id).subscribe(response => {
       let service = response.data;
-      this.provider = response.data.deployment_accounts[0].provider
+      if(response.data.deployment_accounts){
+        this.provider = response.data.deployment_accounts[0].provider
+      } else {
+        // core services will always be in aws
+        this.provider = 'aws'
```
Comment (devsatishm) [inline core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts]: Let core services specify provider value during service creation. Let's avoid these assumptions if possible. Remove `if.. else` block.

File: core/jazz_ui/src/app/pages/services-list/services-list.component.ts (hunk 1 above)
Comment (devsatishm) [reply core/jazz_ui/src/app/pages/services-list/services-list.component.ts]: No, service should have deployment_accounts as part of the service details. Remove `if` block completely.

File: core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.html [hunk 6]
```diff
@@ -50,9 +50,8 @@ <h4 class="bold-title">
                     <section class="service-box-section align-self">OR</section>
                     <section class="pad-left0 service-box-section">
                         <div tabindex="0" class="service-box" [ngClass]="{'service-box-selected': typeOfService == 'sls-app'}" (keyup.enter)="changeServiceType('sls-app')" (click)="changeServiceType('sls-app')">
-                            <div class="icon-icon-sls-app" [ngClass]="{'selected': typeOfService == 'sls-app'}">
-                                <img *ngIf="typeOfService == 'sls-app'" src="./../../../assets/images/icons/Build-custom.svg"/>
-                                <img *ngIf="typeOfService != 'sls-app'" src="./../../../assets/images/icons/Build-custom-gray.svg"/>
+                            <div class="icon-icon-sls-app selected">
+                                <img src="./../../../assets/images/icons/Build-custom.svg"/>
```
Comment (devsatishm) [inline core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.html]: How are these `img` blocks handled? I see them for this & `function`, but they don't exist for `api` & `website` types. Can we make this consistent?

File: core/jazz_ui/src/app/pages/services-list/services-list.component.ts (hunk 1 above)
Comment (devsatishm) [reply core/jazz_ui/src/app/pages/services-list/services-list.component.ts]: No, we should remove `if` check completely. service is expected to contain this information anyways (required field during service creation).

File: core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.html (hunk 6 above)
Comment (devsatishm) [reply core/jazz_ui/src/app/secondary-components/create-service/oss/create-service.component.html]: Discussed with @pgobin, we'll keep this for now since we are not introducing this change now.

File: core/jazz_ui/src/app/pages/services-list/services-list.component.ts (hunk 1 above)
Comment (SauarvD) [reply core/jazz_ui/src/app/pages/services-list/services-list.component.ts]: @raghits @devsatishm It is actually for the core services which are created during stack creation. I checked in the azure as well as sls-app stack. The core services doesn't have the deployment_accounts field. And for azure, now we have a new field in service list page called platform and since core service are missing the required field, so when in UI I open the any core services UI breaks. That's why this change.

File: builds/jenkins-build-pack-api/Jenkinsfile (hunk 2 above)
Comment (SauarvD) [reply builds/jenkins-build-pack-api/Jenkinsfile]: Actually during resolving merge conflicts with develop, accidentally this line got duplicated and I missed it. So I removed this here since it's already present just few lines above the code.

File: core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts (hunk 4 above)
Comment (SauarvD) [reply core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts]: It is actually for the core services which are created during stack creation. I checked in the azure as well as sls-app stack. The core services doesn't have the deployment_accounts field. And for azure, now we have a new field in service list page called platform and since core service are missing the required field, so when in UI I open the any core services UI breaks. That's why this change.

File: core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts (hunk 5 above)
Comment (SauarvD) [reply core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts]: Since core services doesn't have the required deployment_accounts field, we might need to make other changes for specifying the provider for core services during stack installation

File: builds/service-onboarding-build-pack/Jenkinsfile (hunk 3 above)
Comment (SauarvD) [reply builds/service-onboarding-build-pack/Jenkinsfile]: Done

File: core/jazz_ui/src/app/pages/services-list/services-list.component.ts (hunk 1 above)
Comment (devsatishm) [reply core/jazz_ui/src/app/pages/services-list/services-list.component.ts]: I think we have discussed about the `platform` field. There is no need to introduce this field & we should use `deployment_accounts` information to identify the provider information. If this requires installer changes (send `deployment_accounts` as part of bootstrapping for core services), we should make those changes in the installer.

File: builds/jenkins-build-pack-api/Jenkinsfile (hunk 2 above)
Comment (devsatishm) [reply builds/jenkins-build-pack-api/Jenkinsfile]: Yep, just checked. Resolving this.

File: core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts (hunk 4 above)
Comment (devsatishm) [reply core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts]: Same as above - Please see https://github.com/tmobile/jazz/pull/1156#discussion_r326942625

File: core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts (hunk 5 above)
Comment (devsatishm) [reply core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts]: Yes, we need this change at the installer side (if not being done already).

File: core/jazz_ui/src/app/pages/services-list/services-list.component.ts (hunk 1 above)
Comment (SauarvD) [reply core/jazz_ui/src/app/pages/services-list/services-list.component.ts]: Done. Removed the if else block. Services are expected to have now deployment_accounts field

File: core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts (hunk 4 above)
Comment (SauarvD) [reply core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts]: Done. Removed the if else block. Services are expected to have now deployment_accounts field

File: core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts (hunk 5 above)
Comment (SauarvD) [reply core/jazz_ui/src/app/pages/service-detail/service-detail.component.ts]: Done. Removed the if else block. Services are expected to have now deployment_accounts field
